### PR TITLE
Friends

### PR DESCRIPTION
**Pull request by FordMcLeod**
Accept me! I enable ~ friendship ~ and start FOAF

### CODE REVIEW

File: posts/tests/test_views.py [hunk 1]
```diff
@@ -267,3 +310,226 @@ def test_create_valid_other_comment(self):
...
+        force_authenticate(request, user=user)
+        response = view(request)
+        follow = Follow.objects.get(follower=user,followee=other)
+        followRequest = FollowRequest.objects.get(requester=user,requestee=other)
+        self.assertIsNotNone(follow)
+        self.assertIsNotNone(followRequest)
+        self.assertEqual(follow,response.data['follow'])
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_friend_request__follow(self):
```
Comment (MichaelParadis) [inline posts/tests/test_views.py]: Remove the extra _

File: posts/views.py [hunk 2]
```diff
@@ -37,8 +38,161 @@ def put(self, request):
...
+            return False
+
+    def post(self, request):
+        user = request.data.get("author")
+        other = request.data.get("friend")
+        if user is not None and other is not None:
+            try:
+                user_obj = User.objects.get(id=user['id'])
+            except User.DoesNotExist:
+                return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (MichaelParadis) [inline posts/views.py]: I think these should be 400 instead.

File: posts/views.py [hunk 3]
```diff
@@ -37,8 +38,161 @@ def put(self, request):
...
+            try:
+                other_obj = User.objects.get(id=other['id'])
+            except User.DoesNotExist:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+        followSerializer = FollowSerializer(data=request.data, context={'create':True,'followee':other,'follower':user})
```
Comment (MichaelParadis) [inline posts/views.py]: Remove create context. It is not needed

**Review by MichaelParadis — CHANGES_REQUESTED** [review]
Sorry for not actually making this earlier. Let me know when this is good,

**Comment by MichaelParadis** [comment]
@FordMcLeod you have some conflicts. Can you please fix these so we can merge.

File: posts/views.py [hunk 4]
```diff
@@ -351,41 +351,41 @@ def get(self, request, post_id):
...
+# 	    "authors": claimedFriends
+#     }
+#     url = str(userid)+"/friends/"
+#     # Make POST request to url, with userData above, 
+
+#     bridges= response.data['authors']
+
+#     authorData = {
+# 	    "query":"friends",
+# 	    "author":userid,
```
Comment (MichaelParadis) [inline posts/views.py:369]: Is there a reason for this being commented out?

**Review by MichaelParadis — COMMENTED** [review]
Looks fine if the commenting out is intended please SQUASH AND MERGE so that we can put an end to this rebase mess that happened.

File: posts/views.py (hunk 4 above)
Comment (FordMcLeod) [reply posts/views.py:369]: It currently doesn't make a difference whether it is implemented is not. I just commented it out bc it's not being used atm

**Review by MichaelParadis — APPROVED** [review]

File: posts/views.py (hunk 4 above)
Comment (MichaelParadis) [reply posts/views.py:369]: Ok

**Pull request merged**